qiita_pet/handlers/api_proxy/artifact.py

```diff
@@ -5,8 +5,78 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
+from qiita_core.util import execute_as_transaction
+from qiita_pet.handlers.api_proxy.util import check_access, check_fp
 from qiita_db.artifact import Artifact
-from qiita_pet.handlers.api_proxy.util import check_access
+from qiita_db.metadata_template.prep_template import PrepTemplate
+from qiita_db.util import convert_to_id
+
+
+@execute_as_transaction
+def artifact_post_req(user_id, filepaths, artifact_type, name,
+                      prep_template_id):
+    """Creates the prep template and initial artifact for the prep template
+
+    Parameters
+    ----------
+    user_id : str
+        User adding the atrifact
+    filepaths : dict of {str: [str, ...], ...}
+        List of files to attach to the artifact, keyed to the file type
+    }
+    artifact_type : str
+        The type of the artifact
+    name : str
+        Name to give the artifact
+    prep_template_id : int
+        Prep template to attach the artifact to
+
+    Returns
+    -------
+    dict of objects
+        A dictionary containing the new artifact ID
+        {'status': status,
+         'message': message,
+         'artifact': id}
+    """
+    prep = PrepTemplate(int(prep_template_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+    study_id = PrepTemplate.study_id
+    cleaned_filepaths = []
+    for ftype in filepaths:
+        # Convert filepath type to the database ID for adding artifact
+        fp_id = convert_to_id(ftype, 'filepath_type')
```

Comment:
It looks like the documentation for the artifact creation is a bit off (as a result of the documentation of the function `insert_filepaths` being also off). This line is not needed. If you pass the string, it internally changes it to the id, so it is cleaner here if you don't change to the id here (so the interface code is decouples from qiita_db.